Automattic/pocket-casts-android
# [Give rating] Update rating

## Description
- Uses `/user/podcast_rating/add` to update a podcast that has previous rate
- Uses `/user/podcast_rating/show` to fetch the individual rate
- Disable submit button when does not have a rate selected
- Hide Submit button when the the previous rate and current rate selected to update is the same
- 41d99557e95289dbbcdf0f4b324a8a123c5d1fe4 I organized the UI state check


Tracks will be added here: https://github.com/Automattic/pocket-casts-android/pull/2459

## Testing Instructions
1. Open a podcast that you did not listen any episode
2. Try to rate a podcast that you did not rate before
3. ✅ Ensure the rating screen is opened with submit button disabled
4. Select any rate
5. ✅ Ensure the submit button is enabled
6. Rate the podcast
7. Open the rating screen again
8. ✅ Ensure you see the rate you gave in step 6 and submit button is hidden
9. Select a different rate
10. ✅ Ensure submit button is visible

## Screenshots or Screencast 
[Screen_recording_20240710_141131.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/db2af6bf-8bef-4499-845e-e4703f8adc6d)


## Checklist
 
#### I have tested any UI changes...
